plugins/inputs/influxdb_v2_listener/influxdb_v2_listener.go

```diff
@@ -0,0 +1,396 @@
+package influxdb_v2_listener
+
+import (
+	"compress/gzip"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/internal"
+	tlsint "github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/parsers/influx"
+	"github.com/influxdata/telegraf/selfstat"
+)
+
+const (
+	// defaultMaxBodySize is the default maximum request body size, in bytes.
+	// if the request body is over this size, we will return an HTTP 413 error.
+	defaultMaxBodySize = 32 * 1024 * 1024
+)
+
+type InfluxDBV2Listener struct {
+	ServiceAddress string `toml:"service_address"`
+	port           int
+	tlsint.ServerConfig
+
+	ReadTimeout  internal.Duration `toml:"read_timeout"`
+	WriteTimeout internal.Duration `toml:"write_timeout"`
+	MaxBodySize  internal.Size     `toml:"max_body_size"`
```

Comment:
Not sure we really need a max body size. If we do, 32mb is possibly too small. We should be streaming the request through, so this won't affect memory used.

Reply:
ah, yeah - good catch, InfluxDB 2.x now no longer has a `max-body-size` config parameter - https://v2.docs.influxdata.com/v2.0/reference/config-options/

Reply:
I would keep this option, 32mb seems like plenty. The typical batch size is 1000-5000 metrics and is gzip compressed.

Reply:
The limit seems pretty arbitrary if we're taking results of any size in a stream.

Reply:
This API from InfluxDBv2 is built around accepting batches of data, it's not a streaming API.  As everything is loaded into memory, I think this is is a useful limit to avoid OOM.